Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseCollectionStateManager.cs

```diff
@@ -180,6 +233,37 @@ public TimeSpan UpdateState(string instrumentationKey)
             return this.DetermineBackOffs();
         }
 
+        private void UpdateConfiguration(CollectionConfigurationInfo configurationInfo)
+        {
+            // we only get here if Etag in the header is different from the current one, but we still want to check if Etag in the body is also different
+            if (configurationInfo != null && !string.Equals(configurationInfo.ETag, this.currentConfigurationETag, StringComparison.Ordinal))
+            {
+                this.collectionConfigurationErrors.Clear();
+
+                CollectionConfigurationError[] errors = null;
+                try
+                {
+                    errors = this.onUpdatedConfiguration?.Invoke(configurationInfo);
+                }
+                catch (Exception e)
+                {
+                    this.collectionConfigurationErrors.Add(
+                        CollectionConfigurationError.CreateError(
+                            CollectionConfigurationErrorType.CollectionConfigurationFailureToCreateUnexpected,
+                            string.Format(CultureInfo.InvariantCulture, "Unexpected error applying configuration. ETag: {0}", configurationInfo.ETag ?? string.Empty),
+                            e,
+                            Tuple.Create("ETag", configurationInfo.ETag)));
+                }
+
+                if (errors != null)
+                {
+                    this.collectionConfigurationErrors.AddRange(errors);
+                }
+
+                this.currentConfigurationETag = configurationInfo.ETag;
+            }
```

Comment:
How service will reset the configuration and keep default collection only? Will it send "empty" configuration? 

What if service crashed and forgot all etags? 

I think the default behavior should be to stop collecting any custom configuration. Otherwise it SDK may stuck with the bad config.